examples/quickstart/notebooks/quickstart.ipynb

```diff
@@ -544,111 +531,119 @@
+    "the link mentioned after the \"Listening at:\" `INFO` log statement."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {
-    "scrolled": false
-   },
+   "metadata": {},
    "outputs": [],
    "source": [
-    "@pipeline(enable_cache=False)\n",
-    "def training_pipeline():\n",
-    "    \"\"\"Train, evaluate, and deploy a model.\"\"\"\n",
-    "    X_train, X_test, y_train, y_test = training_data_loader()\n",
-    "    model = tree_trainer_mlflow(X_train=X_train, y_train=y_train)\n",
-    "    evaluator(X_test=X_test, y_test=y_test, model=model)\n",
-    "    mlflow_register_model_step.with_options(\n",
-    "        parameters=dict(\n",
-    "            name=\"zenml-quickstart-model\",\n",
-    "            metadata=ModelRegistryModelMetadata(arch=\"decision_tree\"),\n",
-    "            description=\"The second run of the Quickstart pipeline.\",\n",
-    "        )\n",
-    "    )(model)\n",
+    "import os\n",
+    "from zenml.integrations.mlflow.mlflow_utils import get_tracking_uri\n",
+    "\n",
+    "os.environ[\"MLFLOW_TRACKING_URI\"] = get_tracking_uri()\n",
     "\n",
-    "training_pipeline()"
+    "if Environment.in_google_colab():\n",
+    "    # run mlflow through a cloudflare tunnel to get a public endpoint\n",
+    "    !mlflow ui --backend-store-uri $MLFLOW_TRACKING_URI & cloudflared tunnel --url http://localhost:5000\n",
+    "else:\n",
+    "    !mlflow ui --backend-store-uri $MLFLOW_TRACKING_URI"
    ]
   },
   {
    "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Notice that the second pipeline ran slightly faster than the first? That's because ZenML understands that the `data_loader` step of your pipeline is unchanged, so it just reloads the output from your previous run and goes straight to the trainer part. This saves valuable time as you iterate on your pipeline."
+    "Our pipeline above registered the best model with the MLflow model registry.\n",
+    "Whenever you register a model it also versions the model since it's likely that\n",
+    "you'll be iterating and improving your model over time.\n",
+    "\n",
+    "We'll now turn to actually deploying our model and serving some predictions, for\n",
+    "which we'll need to specify the model version we want to use. You can specify\n",
+    "the version number manually but below we'll use the ZenML `Client` to get the\n",
+    "latest version number."
    ]
   },
   {
-   "attachments": {},
-   "cell_type": "markdown",
+   "cell_type": "code",
+   "execution_count": null,
    "metadata": {},
+   "outputs": [],
    "source": [
-    "## Run inference pipeline to deploy and inference on the registered model"
+    "from zenml.client import Client\n",
+    "\n",
+    "most_recent_model_version_number = int(\n",
+    "    Client()\n",
+    "    .active_stack.model_registry.list_model_versions(metadata={})[0]\n",
+    "    .version\n",
+    ")\n",
+    "most_recent_model_version_number"
```

Comment:
Minor nitpick but this looks a bit weird, maybe could be more readable if split into two lines

```python
versions = Client().active_stack.model_registry.list_model_versions()
most_recent_model_version_number = int(versions[0])
```

Reply:
Unfortunately the metadata is a required input, even if it's empty. I'll make a ticket, but @safoinme confirmed that we can make it optional.

Reply:
https://zenml.atlassian.net/browse/FRW-2163